### DISCUSSION TITLE
Am I doing something wrong or does smart cut not work with mkv?

### DISCUSSION THREAD

**Discussion by NadCat1** [mifi/lossless-cut#1292] [Support]
Keyframe cutting works flawlessly and I especially like that subtitles aren't deleted unlike with some other programs. However, I've not once managed to use smart cut or normal cut with mkv files (at least not when cutting something from the beginning). 

Smart cut results in the edited video starting with a ton of artifacts, then plays a part of the cut sequence, and then plays the rest of the video as soon as the next keyframe is hit. That usually even results in the shortened video being longer than the original one.

Normal cut doesn't duplicate anything but the video is black until the next keyframe. Audio seems unaffected and cuts as expected.

Both seem to work pretty well with mp4 files. Is the only solution to convert everything to mp4 before cutting?

**Comment by mifi** [comment]
Smart cut is very experimental so it doesn't work with all files, and there's not much I can do about it unfortunately. Hope for better support in future ffmpeg versions.

But if what you're saying is correct, that converting files from mkv to mp4 (using losslesscut) before smart cutting, and it works, then that's an interesting workaround. Can you confirm that that really works? What kind of codecs are you smart cutting?

> Normal cut doesn't duplicate anything but the video is black until the next keyframe. Audio seems unaffected and cuts as expected.

this is expected with "normal cut". I will explain in better in the main readme.

**Reply by MoralCode** [reply]
> But if what you're saying is correct, that converting files from mkv to mp4 (using losslesscut) before smart cutting, and it works, then that's an interesting workaround. Can you confirm that that really works?

that appears to be the case for me, thanks for the idea to give this a try! this will likely save me literally hours of transcoding several near-8-hour livestreams from h264/mkv to vp9/webm when losslesscut can handle smart cut with h264/mp4.

**Reply by mifi** [reply]
ok, cool i will add it to troubleshooting!

**Comment by NadCat1** [comment]
I'm afraid that was not what I was trying to say, I just noticed that smart cut worked with mp4. I just tried converting with losslesscut before using smart cut and it kinda worked? There's still a small hiccup with a second or so of audio repeating and the video stopping but what was cut from the beginning stays cut and there aren't any noticeable artifacts. I also tried it with AVC (with a very heavy quality loss) and that seems to work flawlessly. I'd assume adding an option to reencode as mp4 while trying to minimize quality loss could be a workaround

**Comment by songogo** [comment]
为什么音频丢失了，怎么设置不会损坏源文件

Why is the audio lost? How can I set it up so that the source file will not be damaged?
